public/javascripts/systems.js

```diff
@@ -1,61 +1,147 @@
+      element.addEventListener('click', function() {SensorDropdown(s);});
+      element.style['cursor'] = 'pointer';
+    }
   }));
+  properties = [];
+  for (var property of metadata.getElementsByTagName('property')) {
+    properties.push({
+        'sensor': property.getAttribute('sensor'),
+        'targetId': property.getAttribute('targetId'),
+        'targetAttribute': property.getAttribute('targetAttribute'),
+        'scaling': parseFloat(GetAttributeOrDefault(property, 'scaling', 1)),
+        'offset': parseFloat(GetAttributeOrDefault(property, 'offset', 0)),
+        'min': parseFloat(GetAttributeOrDefault(property, 'min', -Infinity)),
+        'max': parseFloat(GetAttributeOrDefault(property, 'max', Infinity)),
+    });
+  }
+  linktargets = {};
+  for (var link of links) {
+    for (var element of doc.querySelectorAll(`[id^=link_${link}]`)) {
+      linktargets[element.id] = link;
+      element.addEventListener('click', LoadSVG);
+      element.style['cursor'] = 'pointer';
+    }
+  }
   UpdateOnce();
 }
 
 function LoadSVG(fn) {
-  if (typeof fn != 'undefined') {
-    var current_fn = $("#svg_frame").attr('data');
-    var p = current_fn.split('/');
-    if (fn.slice(-4) != '.svg')
-      fn = `${fn}.svg`;
-    p[p.length-1] = fn;
-    fn = p.join('/');
-    $("#svg_frame").attr('data', fn);
-  } else {
+  try {
+    // Need to determine new SVG based on event target
+    // unless fn is already a string, then get exception
+    fn = linktargets[fn.currentTarget.id];
+  } catch (e) {
+    // Was probably a string. Do nothing.
   }
+  //if (typeof fn != 'undefined') {
+  var current_fn = $("#svg_frame").attr('data');
+  var p = current_fn.split('/');
+  if (fn.slice(-4) != '.svg')
+    fn = `${fn}.svg`;
+  p[p.length-1] = fn;
+  fn = p.join('/');
+  $("#svg_frame").attr('data', fn);
+  //}
   console.log(`Loading ${$("#svg_frame").attr('data')}`);
-  setTimeout(() => {
-    // this is jank but not sure how else to do this
-    Setup();
-    console.log('Setup?');
-  }, 500); // if loading takes more than 500ms then increase
 }
 
 function UpdateOnce() {
+  var updatestarttime = Date.now();
   var doc = document.getElementById('svg_frame').getSVGDocument();
   sensors.forEach(s => {
     $.getJSON(`/devices/get_last_point?sensor=${s}`, data => {
-      if (valves.includes(s)) {
-        var elements = doc.querySelectorAll(`[id^=valve_${s}]`);
-        for (var element of elements) {
-          element.classList.remove(data.value ? 'off' : "on");
-          element.classList.add(data.value ? 'on' : 'off');
+      var value = parseFloat(data.value);
+      for (var element of doc.querySelectorAll(`[id^=valve_${s}]`)) {
+        element.classList.remove(value ? 'off' : "on");
+        element.classList.add(value ? 'on' : 'off');
+      }
+      for (var element of doc.querySelectorAll(`[id^=value_${s}]`)) {
+        element.innerHTML = `${SigFigs(value)}`;
+      }
+      for (var property of properties) {
```

Comment:
Would it make sense to make `properties` an object rather than array, so you could just index `properties[sensor]` rather than looping through to find the one that matches?

Reply:
Well I certainly cannot argue when you throw my own comment on another pull request back at me 😁 Of course, an object would make more sense